### PR TITLE
ramips: linkit: replace driver with gpio-hog

### PR DESCRIPTION
**Pull request by neheb**
It does the same thing minus a few dmesg prints.

ping @DragonBluep

### CODE REVIEW

File: target/linux/ramips/dts/mt7628an_mediatek_linkit-smart-7688.dts [hunk 1]
```diff
@@ -144,6 +138,15 @@
 	};
 };
 
+&gpio {
+	bootstrap {
+		gpio-hog;
+		line-name = "bootstrap";
+		output-low;
+		gpios = <11 GPIO_ACTIVE_HIGH>;
```
Comment (DragonBluep) [inline target/linux/ramips/dts/mt7628an_mediatek_linkit-smart-7688.dts:146]: It's better to use `GPIO_ACTIVE_LOW`, though the GPIO status is controlled by `output-low`/`output-high` property.

**Review by DragonBluep — APPROVED** [review]
More details: https://github.com/openwrt/openwrt/pull/16766.

File: target/linux/ramips/dts/mt7628an_mediatek_linkit-smart-7688.dts (hunk 1 above)
Comment (neheb) [reply target/linux/ramips/dts/mt7628an_mediatek_linkit-smart-7688.dts:146]: right, I was confused by this. Other usages of gpio-hog are inconsistent as well.

**Comment by robimarko** [comment]
Thanks! Rebased on top of main and merged!

**Pull request merged**